### PR TITLE
Adds logger id for dynamic_modules

### PR DESCRIPTION
**Pull request by mathetake**
Commit Message: Adds logger id for dynamic_modules
Additional Description:

This commit adds a logger_id for the core dynamic_modules extension.

* https://github.com/envoyproxy/envoy/tree/main/source/extensions/dynamic_modules
* https://github.com/envoyproxy/envoy/tree/main/source/extensions/filters/http/dynamic_modules

This is to mainly avoid unnecessarily requesting review from logger owners into
dynamic_modules PRs just for adding one line.

Risk Level: low
Testing: none
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/common/common/logger.h [hunk 1]
```diff
@@ -50,6 +50,7 @@ const static bool should_log = true;
   FUNCTION(decompression)                                                                          \
   FUNCTION(dns)                                                                                    \
   FUNCTION(dubbo)                                                                                  \
+  FUNCTION(dynamic_modules)                                                                        \
```
Comment (wbpcode) [inline source/common/common/logger.h:53]: Could you add this at the end of the list? This change all these enum's value which may bring potential risk.

**Review by wbpcode — COMMENTED** [review]
LGTM overall except only one comment.

File: source/common/common/logger.h (hunk 1 above)
Comment (mathetake) [reply source/common/common/logger.h]: https://github.com/envoyproxy/envoy/pull/37296/commits/34af90e8a0216a32b9ff26335420b081070b86e6 done

**Review by wbpcode — APPROVED** [review]

**Comment by wbpcode** [comment]
hah, you changed you head icon. 😃

**Pull request merged**